src/db_options.rs

```diff
@@ -399,39 +399,6 @@ impl Options {
         }
     }
 
-    /// Hints to the OS that it should not buffer disk I/O. Enabling this
-    /// parameter may improve performance but increases pressure on the
-    /// system cache.
-    ///
-    /// The exact behavior of this parameter is platform dependent.
-    ///
-    /// On POSIX systems, after RocksDB reads data from disk it will
-    /// mark the pages as "unneeded". The operating system may - or may not
-    /// - evict these pages from memory, reducing pressure on the system
-    /// cache. If the disk block is requested again this can result in
-    /// additional disk I/O.
-    ///
-    /// On WINDOWS systems, files will be opened in "unbuffered I/O" mode
-    /// which means that data read from the disk will not be cached or
-    /// bufferized. The hardware buffer of the devices may however still
-    /// be used. Memory mapped files are not impacted by this parameter.
-    ///
-    /// Default: true
-    ///
-    /// # Example
-    ///
-    /// ```
-    /// use rocksdb::Options;
-    ///
-    /// let mut opts = Options::default();
-    /// opts.set_allow_os_buffer(false);
-    /// ```
-    #[deprecated(since="0.7.0", note="replaced with set_use_direct_reads/set_use_direct_writes methods")]
-    pub fn set_allow_os_buffer(&mut self, is_allow: bool) {
```

Comment:
Ahh, looks like there's two issues there:

```
---- src/db_options.rs - db_options::Options::set_allow_os_buffer (line 423) stdout ----
	warning: use of deprecated item: replaced with set_use_direct_reads/set_use_direct_writes methods
 --> <anon>:6:6
  |
6 | opts.set_allow_os_buffer(false);
  |      ^^^^^^^^^^^^^^^^^^^
  |
  = note: #[warn(deprecated)] on by default
```
This is a warning because I deprecated that method but forgot to add ``#[allow(deprecated)]`` to its doctest.

```
error: linking with `cc` failed: exit code: 1
  |
  = note: "cc" "-Wl,--as-needed" "-Wl,-z,noexecstack" "-m64" "-L" "/home/rdodd/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "/tmp/rustdoctest.yFOM2FA1OKGb/rust_out.0.o" "-o" "/tmp/rustdoctest.yFOM2FA1OKGb/rust_out" "/tmp/rustdoctest.yFOM2FA1OKGb/rust_out.crate.allocator.o" "-Wl,--gc-sections" "-pie" "-nodefaultlibs" "-L" "/home/rdodd/.cargo/target_dir/debug/deps" "-L" "/home/rdodd/.cargo/target_dir/debug/build/librocksdb-sys-ff15d401df6cb8a6/out" "-L" "/home/rdodd/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-Wl,-Bstatic" "/home/rdodd/.cargo/target_dir/debug/deps/librocksdb-e312e09db9ca21d5.rlib" "/home/rdodd/.cargo/target_dir/debug/deps/liblibrocksdb_sys-ae3566a82a4c0e36.rlib" "/home/rdodd/.cargo/target_dir/debug/deps/liblibc-d2710bfb355676bc.rlib" "-L" "/home/rdodd/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-Wl,-Bdynamic" "-l" "std-1d2a102a93f1dfa9" "-Wl,-Bstatic" "/home/rdodd/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcompiler_builtins-821b4ec553d97f1e.rlib" "-Wl,-Bdynamic" "-l" "stdc++" "-l" "util" "-l" "dl" "-l" "rt" "-l" "pthread" "-l" "pthread" "-l" "gcc_s" "-l" "c" "-l" "m" "-l" "rt" "-l" "pthread" "-l" "util"
  = note: /home/rdodd/.cargo/target_dir/debug/deps/librocksdb-e312e09db9ca21d5.rlib(rocksdb-e312e09db9ca21d5.0.o): In function `rocksdb::db_options::{{impl}}::set_use_direct_writes':
          /home/rdodd/devel/rust-rocksdb/src/db_options.rs:398: undefined reference to `rocksdb_options_set_use_direct_writes'
          collect2: error: ld returned 1 exit status
```

This is a separate issue because RocksDB has removed the ``rocksdb_options_set_use_direct_writes`` function